lib/base_feed.js

```diff
@@ -0,0 +1,192 @@
+'use strict';
+
+
+const P = require('bluebird');
+const HyperSwitch = require('hyperswitch');
+const uuid = require('cassandra-uuid').TimeUuid;
+const mwUtil = require('./mwUtil');
+const URI = HyperSwitch.URI;
+
+const DEFAULT_TTL = 3600;
+const DEFAULT_CACHE_CONTROL = 's-maxage=30, max-age=15';
+
+function constructBody(result) {
+    const body = {};
+    Object.keys(result).forEach((key) => {
+        if (result[key].body && Object.keys(result[key].body).length) {
+            body[key] = result[key].body;
+        }
+    });
+    return body;
+}
+
+class BaseFeed {
+    constructor(options) {
+        options.ttl = options.ttl || DEFAULT_TTL;
+        options.feed_cache_control = options.feed_cache_control || DEFAULT_CACHE_CONTROL;
+        if (!options.host) {
+            throw new Error('feed module: host option missing');
+        }
+
+        this.options = options;
+    }
+
+    _makeFeedRequests(parts, hyper, rp, dateArr) {
+        const props = {};
+        parts.forEach((part) => {
+            const def = this.options.part_uris[part];
+            const uriArray = [this.options.host, rp.domain].concat(def.uri);
+            if (def.date) {
+                Array.prototype.push.apply(uriArray, dateArr);
+            }
+            props[part] = hyper.get({
+                uri: uriArray.join('/'),
+                query: { aggregated: true }
```

Comment:
@d00rman Updated. I've decided not to abstract the actual request to the MCS, this could be better done with inheritance then by setting complex parameter